src/main/kotlin/no/nav/familie/ef/sak/økonomi/TilkjentYtelseService.kt

```diff
@@ -18,28 +19,26 @@ import java.util.*
 @Service
 class TilkjentYtelseService(private val økonomiKlient: ØkonomiKlient,
                             private val tilkjentYtelseRepository: TilkjentYtelseRepository,
-                            private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository) {
+                            private val customRepository: CustomRepository<TilkjentYtelse>) {
 
     @Transactional
     fun opprettTilkjentYtelse(tilkjentYtelseDTO: TilkjentYtelseDTO): UUID {
         tilkjentYtelseDTO.valider()
 
-        val eksisterendeTilkjentYtelse = tilkjentYtelseRepository.findByPersonIdentifikatorOrNull(tilkjentYtelseDTO.søker)
-        if (eksisterendeTilkjentYtelse != null)
+        val eksisterendeTilkjentYtelse = tilkjentYtelseRepository.findByPersonident(tilkjentYtelseDTO.søker)
+        if (eksisterendeTilkjentYtelse != null) {
             error("Søker har allerede en tilkjent ytelse")
+        }
 
-        val opprettetTilkjentYtelse = tilkjentYtelseRepository.save(
-                tilkjentYtelseDTO.tilTilkjentYtelse(TilkjentYtelseStatus.OPPRETTET)
-        )
-
-        andelTilkjentYtelseRepository.saveAll(tilkjentYtelseDTO.tilAndelerTilkjentYtelse(opprettetTilkjentYtelse.id))
+        val opprettetTilkjentYtelse =
+                customRepository.persist(tilkjentYtelseDTO.tilTilkjentYtelse(TilkjentYtelseStatus.OPPRETTET))
```

Comment:
Hva er forskjellen på `customRepository.persist` og `tilkjentYtelseRepository.save` i denne sammenheng?

Reply:
Save prøver å oppdatere forekomsten hvis den har primærnøkkel. Side forkomsten har primærnøkkel så kjører sping en update where som ikke feiler, men som heller ikke gjør noe.

Reply:
customRepository oppdaterer vel også relationer til objektet. Eks hvis du oppretter
Car -> wheel, og tar persist på car, så opprettes (listen med) wheels også

Reply:
Save oppdaterer også relasjonene.

Reply:
Det virket ikke for meg i SakRepositoryTest, hvis jag brukte `sakRepository.save` ble ikke relationerna opprettet